### PR TITLE
Swagger for the new ARM resource PrivateLinkHub

### PR DESCRIPTION
**Pull request by sugandhaamic**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/202/Overall-Process-of-AME-Onboarding) for the workflow.
- [x] [Validation tools](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/84/Swagger-Validation-tools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/FAQ.md).

### ARM API Review Checklist
- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process]( https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/212/Swagger-PR-Review).

### CODE REVIEW

**Comment by raych1** [comment]
@sugandhaamic , please fix the CI check failures.

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/privatelinkhub.json [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "schema": {
+              "$ref": "#/definitions/PrivateLinkHubInfoListResult"
+            },
+            "description": ""
```
Comment (raych1) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/privatelinkhub.json:240]: Can you provide more text for the `description`?

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/privatelinkhub.json [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+            "description": "OK"
+          },
+          "204": {
+            "description": ""
+          },
+          "default": {
+            "schema": {
+              "$ref": "#/definitions/ErrorContract"
+            },
+            "description": ""
```
Comment (raych1) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/privatelinkhub.json:205]: Same here to add more text for the `description`.

**Comment by sugandhaamic** [comment]
> @sugandhaamic , please fix the CI check failures.

CI issues fixed

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/privatelinkhub.json (hunk 1 above)
Comment (sugandhaamic) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/privatelinkhub.json]: Description added

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/privatelinkhub.json (hunk 2 above)
Comment (sugandhaamic) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/privatelinkhub.json]: Description added

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/privatelinkhub.json [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+          },
+          "type": "object"
+        }
+      }
+    },
+    "PrivateLinkHubProperties": {
+      "description": "PrivateLinkHub properties",
+      "type": "object",
+      "properties": {
+        "provisioningState": {
```
Comment (ravbhatnagar) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/privatelinkhub.json:400]: please make it an enum and add all possible values.

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/privatelinkhub.json [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+        "tags": {
+          "description": "Resource tags",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "type": "object"
+        }
+      }
+    },
+    "PrivateLinkHubProperties": {
```
Comment (ravbhatnagar) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/privatelinkhub.json:396]: why does this type not have any other properties? Can you please provide some more info on what this type does?

**Review by ravbhatnagar — COMMENTED** [review]
@sugandhaamic - few comments

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/privatelinkhub.json (hunk 4 above)
Comment (sugandhaamic) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/privatelinkhub.json:396]: The resource does not have any properties so far. This resource is going to be used to create a Private Links to Synapse studio, thats all.

**Review by KrisBash — APPROVED** [review]
Signing off for preview. A resource with no properties is an oddity though.

**Pull request merged**